### DISCUSSION TITLE
No "Toolbars" right-click menu on Win11 Enterprise

### DISCUSSION THREAD

**Discussion by alextfish** [valinet/ExplorerPatcher#2920] [Q&A]
I've been forcibly "upgraded" against my will to Windows 11 Enterprise at work. I need a way to get back my various toolbars (Quick Launch style, but my own directories). I was so happy to discover the reports of this app. But it doesn't seem to work as described for me. When I install ExplorerPatcher I get the Win10-style taskbar - great. I unlock the taskbar and can expand it - great. But I right-click it and I *don't* get the "Toolbars" context menu item. Apparently plenty of other people do, but it's not there for me:
![image](https://github.com/valinet/ExplorerPatcher/assets/414419/7f9baaee-fc8f-41a7-a92e-75494dc050a0)

My machine is Windows 11 Enterprise, version 23H2, OS build 22631.3155. Any ideas why my right-click context menu would not include "Toolbars" after I install ExplorerPatcher?

**Comment by pyrates999** [comment]
Please post windows version and build.  You can find this by running winver.
Please post explorer patcher version.

Please include what settings you changed in explorer patcher properties application.  These should be screen shots.

Please include any other software you have installed that modifies the UI of windows.

**Reply by alextfish** [reply]
I did already post my Windows version and build in the original question...
ExplorerPatcher was downloaded today, so it's whatever today's version is? 22621.2861.62.2, apparently. It also confirms I'm running on Windows 11 OS Build 22631.3155, as I said before.

Nothing else installed that modifies the UI of Windows - as I say, I was just given this Win11 laptop in the past day or two. Dropbox, I guess, and OneDrive and 7-Zip, which all add Explorer context menu items. Obviously MS Edge and MS Copilot are installed (without me asking them to). I haven't done any registry editing. I edited a couple of Windows Explorer settings when I was trying to make Win11 do sensible things (without success), but just the usual "show file extensions" kind of thing.

I haven't changed any ExplorerPatcher settings - I just wanted to add toolbars and it failed at that first hurdle.

**Comment by tallman69** [comment]
In the ExplorerPatcher Properties app (click on `Properties` in the right click context menu) change the Taskbar style to Windows 10.  The right click context menu will then have Toolbars at the top. The Windows 11 style taskbar does not support toolbars.

**Reply by alextfish** [reply]
It's already in Windows 10 style. It went that way as soon as I installed ExplorerPatcher. That's curious, actually, as it says "Taskbar style: Windows 10" and in the same menu "Windows 11" is tagged as "default"; but it isn't. Windows 10 is the default style it applies as soon as you install ExplorerPatcher. I've tried uninstalling and reinstalling it a couple of times.

**Reply by tallman69** [reply]
Have you tried toggling back and forth between the two taskbar styles and restarting File Explorer, using the button on the lower left corner of the Properties window, each time you change the taskbar style?

**Reply by pyrates999** [reply]
EP when it is installed, does not change the taskbar style to windows 10.  You'd have to either change it.  Or someone else previously installed EP on that computer where it had the taskbar set to windows 10.

If that's the case, I would open up the EP properties application and go to **about**.  And then click on **restore default settings**.  Then reboot.

**Reply by joshhu** [reply]
It happened to me after I updated my Windows 11 to `22631.3235`. After Windows update and the machine rebooted, the toolbar was gone. Then I reinstalled ExplorerPatcher, nothing happened. I imported the registry settings I previously exported, and changed the `Taskbar style` to Windows 10. Nothing happened either. But the catch is that "YOU HAVE TO CLICK THE LOWER LEFT 'Restart Fil Explorer`". And after that, the toolbar went back on. 

![image](https://github.com/valinet/ExplorerPatcher/assets/10594453/a4eccd83-6dab-4457-b719-e84491b09987)
![image](https://github.com/valinet/ExplorerPatcher/assets/10594453/7dee49de-3d74-4002-be48-3dea40ac115d)
![image](https://github.com/valinet/ExplorerPatcher/assets/10594453/3bfa52e2-eb85-492e-8626-ed4927c6386e)
![image](https://github.com/valinet/ExplorerPatcher/assets/10594453/b26b8efe-7d31-424a-83f5-bd8a3313834f)

**Reply by alextfish** [reply]
Thanks for the replies. Still not working though. I was able to confirm by various repeated restarts and reinstalls that I don't know what the real default style of EP is, as it leaves its settings through uninstall+reinstall. But no matter how much I restart File Explorer, reinstall EP, Restore Default Settings and/or reboot Windows, I still get the Win11-style context menu just like in the screenshot in the OP, without the toolbar options, even when my taskbar style is set to Windows 10.
![image](https://github.com/valinet/ExplorerPatcher/assets/414419/b23c256b-800a-413b-bdcd-7ec46d98d32b)

Does anyone have any other ideas? Any registry keys I can tweak or delete or anything?

**Reply by alextfish** [reply]
Well, for what it's worth, my laptop got automatically updated to the newest Windows build (bringing [the transparency bug](https://github.com/valinet/ExplorerPatcher/discussions/953)). But somehow it also, finally, finally, brought the Toolbars option to the taskbar's context menu. I still have no idea why it didn't show up through my first several reboots and reinstalls, but it's here now. So... good, I guess?

**Comment by pyrates999** [comment]
See here: https://github.com/valinet/ExplorerPatcher/wiki/All-features#windows-11

**Reply by alextfish** [reply]
I'm afraid that's not helpful. As I've stated, I've got ExplorerPatcher set to Windows 10, which has done _something_ as the taskbar icons are left-aligned rather than centred. I can move the taskbar to top/bottom/left/right using the ExplorerPatcher properties as described in your link in the "Windows 10" section. But I don't have the toolbars submenu in the context menu, which is the only thing I really want.

Or are you trying to say "toolbars aren't listed in the list of All Features, so they're not provided by ExplorerPatcher"? But I see various other people's comments and issues, including on this site, saying that _they_ were able to use ExplorerPatcher to get toolbars on Windows 11, which is the only thing I really want.

**Reply by tallman69** [reply]
>Or are you trying to say "toolbars aren't listed in the list of All Features, so they're not provided by ExplorerPatcher"? 

Toolbars are not a feature of EP, but rather toolbars are a feature of the Windows 10 taskbar which EP allows a person to enable in place of the Windows 11 taskbar.  That is why people say that EP allows them to get toolbars on Windows 11

If you have saved your EP settings changes using the `Export current settings` feature under the `About` tab of the Properties app, you might try deleting that .reg file before reinstalling EP.  If you are comfortable editing the Registry, you might try uninstalling EP and then searching the Registry for any traces of ExplorerPatcher that may be left behind and then removing them before reinstalling EP. But I would use extreme caution and back up your registry before making and changes so the registry can be restored if something goes wrong.

**Reply by Amrsatrio** [reply]
I don't think EP messes anything with turning on/off the toolbars. But I know for sure that there is a policy somewhere in gpedit that allows disabling this.

**Reply by tallman69** [reply]
Here are a couple tutorials that explains how to prevent or allow the addition or removal of toolbars using either Group Policy Editor or Registry Editor.

https://www.isunshare.com/windows-10/cannot-add-or-remove-toolbars-on-taskbar-what-to-do.html

https://www.tenforums.com/tutorials/104386-enable-disable-toolbars-taskbar-windows.html
